src/inmanta/moduletool.py

```diff
@@ -289,10 +283,13 @@ def bump_version(cls, change_type: ChangeType, version: Version, version_tag: st
         If the given version has a different version tag set, it will be ignored.
         """
         parts = [int(x) for x in version.base_version.split(".")]
-        while len(parts) < 3:
+        while len(parts) < 4:
             parts.append(0)
 
+        if change_type is ChangeType.REVISION:
```

Comment:
This comment got a bit out of hand. A summary of what follows below:
- If I understand correctly this currently always results in a four digit version (or I'm missing something). I don't think we want that. This is the only actual hard change request in this comment. Possible fix included in the snippet below but feel free to approach it differently.
- I don't like the big if-else with assignment to each part individually and I realized how we could clean it up. You didn't introduce it, but by adding another branch it got just a bit worse. Not a strict requirement for me but I think we can clean it up without too much effort. PoC below. Feel free to ignore this part if you want.
- I think we should raise an exception if `ChangeType` is none of the expected. Again, not something you introduced, but easy enough to fix it while we're at it. "assert your invariants, even if (or particularly when) they are trivial." - @wouterdb 2022 - https://inmanta.com/2022/02/15/the-test-case-i-wish-i-had-written/

PoC, feel free to rewrite this whole thing if you want.
```python
bump_index: int
if change_type is ChangeType.REVISION:
    bump_index = 3
elif change_type is ChangeType.PATCH:
    bump_index = 2
...
else:
    raise AppropriateException()

base_parts = [int(x) for x in version.base_version.split(".")]
nb_digits: int = max(bump_index + 1, 3, min(len(base_parts), 4)) # use 4th digit only if it already existed or if it is being bumped
parts = list(
    more_itertools.take(
        nb_digits,
        itertools.chain(
            base_parts[:bump_index + 1],
            itertools.repeat([0]),
        ),
    )
)
parts[bump_index] += 1
```

Reply:
The result was always a four-digit version, but if it ended with a 0, I was dropping that number to come back to a three-digit version. But your solution is much nicer than the old one, so I have replaced it with your PoC.

Reply:
Ah, I must've missed the dropping of the fourth digit somehow. Anyway, I guess it's fine now.

Reply:
I see you made a small change to my snippet that is simpler but only valid if the fourth digit does indeed get dropped. Could you point me to where that happens?

Reply:
(The tests are passing so you must be right but I don't see it)

Reply:
And if it does get dropped we might as well simplify it even more by always taking 4 digits.